acp/source_hub_client.go

```diff
@@ -100,6 +103,23 @@ func NewLocalACP() ACP {
 	}
 }
 
+func NewSourceHubACP(
+	chainID string,
+	grpcAddress string,
+	cometRPCAddress string,
+	keyring keyring.Keyring,
```

Comment:
`keyring.Keyring` is an interface? What's problematic about it? And what are you looking for me to do about it in this PR?

Reply:
I'm talking about the dependencies within the keyring package. I also don't think the keyring should be used anywhere other than the CLI. Using the `TxSigner` interface from the sourcehub sdk instead of the keyring would be better imo.

Reply:
> I also don't think the keyring should be used anywhere other than the CLI

I'm quite strongly against that, and it was discussed a while ago: https://discord.com/channels/427944769851752448/1245021339312717824

Reply:
Discussed over discord - I'll see if I can pass the `TxSigner` interface into the ACP module interface or similar.

 - [ ] Revisit direct keyring usage

Reply:
Updated, keyring is no longer referenced internally.

Reply:
Looks really nice!